google-beta/resource_compute_instance_group_manager.go

```diff
@@ -470,8 +492,35 @@ func resourceComputeInstanceGroupManagerUpdate(d *schema.ResourceData, meta inte
 	}
 
 	if d.HasChange("update_policy") {
-		updatedManager.UpdatePolicy = expandUpdatePolicy(d.Get("update_policy").([]interface{}))
-		change = true
+		oldFixed, _ := d.GetChange("update_policy.0.max_surge_fixed")
+		oldPercent, newPercent := d.GetChange("update_policy.0.max_surge_percent")
+		// if we're going from fixed to percent or percent to fixed, we need to use update, not patch
+		// fixed can be set to 0, so to detect percent to fixed, we check if percent was set before and isn't now
+		if (oldFixed.(int) != 0 && newPercent.(int) != 0) || (oldPercent.(int) != 0 && newPercent.(int) == 0) {
+			updatedManager.Name = d.Get("name").(string)
+			updatedManager.Description = d.Get("description").(string)
+			updatedManager.BaseInstanceName = d.Get("base_instance_name").(string)
+			updatedManager.TargetSize = int64(d.Get("target_size").(int))
+			updatedManager.NamedPorts = getNamedPortsBeta(d.Get("named_port").(*schema.Set).List())
+			updatedManager.TargetPools = convertStringSet(d.Get("target_pools").(*schema.Set))
+			if v, ok := d.GetOk("auto_healing_policies"); ok {
+				updatedManager.AutoHealingPolicies = expandAutoHealingPolicies(v.([]interface{}))
+			}
+			updatedManager.Versions = expandVersions(d.Get("version").([]interface{}))
+			updatedManager.UpdatePolicy = expandUpdatePolicy(d.Get("update_policy").([]interface{}))
+			change = false
```

Comment:
change defaults to false already

Reply:
`change` could have been set to `true` by earlier blocks. If we're going to do an `Update`, there's no sense in doing `Patch`es, too, because we're going to write those values in the `Update`, anyways.